spec/jobs/reports/combined_invoice_supplement_report_v2_spec.rb

```diff
@@ -8,341 +8,561 @@
-  let(:inside_iaa2) { iaa2_range.begin + 1.day }
+        let(:user2) { create(:user, profiles: [profile2]) }
+        let(:profile2) { build(:profile, verified_at: DateTime.new(2018, 6, 1).utc) }
 
-  describe '#perform' do
-    it 'is empty with no data' do
-      csv = CSV.parse(report.perform(Time.zone.today), headers: true)
-      expect(csv).to be_empty
-    end
+        let(:user3) { create(:user, profiles: [profile3]) }
+        let(:profile3) { build(:profile, verified_at: DateTime.new(2018, 6, 1).utc) }
 
-    context 'with data' do
-      let(:user1) { create(:user) }
-      let(:user2) { create(:user) }
-      let(:user3) { create(:user) }
-      let(:user4) { create(:user) }
-      let(:user5) { create(:user) }
-      let(:user6) { create(:user) }
-
-      before do
-        iaa_order1.integrations << build_integration(
-          issuer: iaa1_sp.issuer,
-          partner_account: partner_account1,
-        )
-        iaa_order2.integrations << build_integration(
-          issuer: iaa2_sp1.issuer,
-          partner_account: partner_account2,
-        )
-        iaa_order2.integrations << build_integration(
-          issuer: iaa2_sp2.issuer,
-          partner_account: partner_account2,
-        )
-        iaa_order1.save
-        iaa_order2.save
-
-        # iaa1_sp sp_return_logs
-        # 1 unique user in month 1 at IAA 1 sp @ IAL 1
-        create(
-          :sp_return_log,
-          user_id: user1.id,
-          issuer: iaa1_sp.issuer,
-          ial: 1,
-          requested_at: inside_iaa1,
-          returned_at: inside_iaa1,
-          profile_verified_at: nil,
-          billable: true,
-        )
-
-        # 2 new unique users in month 1 at IAA 1 sp @ IAL 2 with profile age 2
-        [user5, user6].each do |user|
-          create(
-            :sp_return_log,
-            user_id: user.id,
-            ial: 2,
+        let(:csv) { CSV.parse(report.perform(Time.zone.today), headers: true) }
+
+        before do
+          iaa_order1.integrations << build_integration(
             issuer: iaa1_sp.issuer,
-            requested_at: inside_iaa1,
-            returned_at: inside_iaa1,
-            profile_verified_at: DateTime.new(2018, 6, 1).utc,
-            billable: true,
+            partner_account: partner_account1,
+          )
+          iaa_order1.save
+
+          # 1 new unique user in month 1 at IAA 1 sp @ IAL 1
+          7.times do
+            create_sp_return_log(
+              user: user1, issuer: iaa1_sp.issuer, ial: 1,
+              returned_at: inside_iaa1
+            )
+          end
+
+          # 2 new unique users in month 1 at IAA 1 sp @ IAL 2 with profile age 2
+          [user2, user3].each do |user|
+            create_sp_return_log(
+              user: user, issuer: iaa1_sp.issuer, ial: 2,
+              returned_at: inside_iaa1
+            )
+          end
+        end
+
+        it 'validates authentication counts in ial1 + ial2 & checks partner single issuer cases' do
```

Comment:
let's use a different word than "validate"? Validate to me means it will throw an error if something is wrong.

```suggestion
        it 'has correct authentication counts in ial1 + ial2 & checks partner single issuer cases' do
```